src/ship.js

```diff
@@ -0,0 +1,28 @@
+const Itinerary = require("./itinerary");
+class Ship {
+  constructor(itinerary) {
+    this.itinerary = itinerary;
+    this.currentPort = itinerary.ports[0];
+    this.previousPort = null;
+    this.currentPort.addShip(this);
+  }
+
+  setSail() {
+    const itinerary = this.itinerary;
+    const currentPortInd = itinerary.ports.indexOf(this.currentPort);
+    if (currentPortInd === this.itinerary.ports.length - 1) {
```

Comment:
on this line you've access `this.itinerary...`, buy you've already saved the `itinerary` to a variable that you could reference here, i.e.
```
if (currentPortInd === itinerary.ports.length - 1) ...
```